py/objtype.c

```diff
@@ -1389,13 +1389,15 @@ STATIC mp_obj_t mp_builtin_isinstance(mp_obj_t object, mp_obj_t classinfo) {
 
 MP_DEFINE_CONST_FUN_OBJ_2(mp_builtin_isinstance_obj, mp_builtin_isinstance);
 
-mp_obj_t mp_instance_cast_to_native_base(mp_const_obj_t self_in, mp_const_obj_t native_type) {
+mp_obj_t mp_instance_cast_to_native_base(mp_obj_t self_in, mp_const_obj_t native_type) {
```

Comment:
With the function as it is now, passing in `native_type` is not needed: the test to see if it's native or not can be written as `mp_obj_is_native_type(self_type)`.

But, since both users of this function do a call to `mp_obj_is_subclass_fast(..., native_type)` to check it's the correct type, maybe it's worth retaining that check here to reduce code size.  For callers that don't need the check they can just pass in MP_OBJ_NULL for the `native_type`, to bypass the check.

Also, if this function now has use for not only instance objects, it should probably be renamed `mp_obj_cast_to_native_base`.

Reply:
OK thanks. It comes out to the same overall code size, but I agree it's neater.

I used `MP_OBJ_FROM_PTR(self_type) == native_type` instead of `mp_obj_is_native_type` as it's a more specific check and it's the same code size. Although it's implied by the is_subclass.

>  For callers that don't need the check they can just pass in MP_OBJ_NULL for the `native_type`, to bypass the check.

As there are no callers who don't require this currently, perhaps that could be added in #5660. (It'll add code size I guess for a small perf benefit)

Reply:
Sounds good!